inventory_management_system_api/migrations/scripts/expected_lifetime_migration.py

```diff
@@ -0,0 +1,211 @@
+    def serialize_url(self, url: HttpUrl):
+        """
+        Convert `url` to string when the model is dumped.
+        :param url: The `HttpUrl` object.
+        :return: The URL as a string.
+        """
+        return url if url is None else str(url)
+
+
+class OldCatalogueItemOut(CreatedModifiedTimeOutMixin, OldCatalogueItemBase):
+    """
+    Output database model for a catalogue item.
+    """
+
+    id: StringObjectIdField = Field(alias="_id")
+    catalogue_category_id: StringObjectIdField
+    manufacturer_id: StringObjectIdField
+    obsolete_replacement_catalogue_item_id: Optional[StringObjectIdField] = None
+    properties: List[PropertyOut] = []
+
+    model_config = ConfigDict(populate_by_name=True, arbitrary_types_allowed=True)
+
+
+# pylint: enable=duplicate-code
+
+
+class Migration(BaseMigration):
+    """Migration for Catalogue Items' Optional Expected Lifetime"""
+
+    description = "Migration for Catalogue Items' Optional Expected Lifetime"
+
+    def __init__(self, database: Database):
+
+        self._catalogue_items_collection: Collection = database.catalogue_items
+
+    def forward(self, session: ClientSession):
+        """This function should actually perform the migration
+
+        All database functions should be given the session in order to ensure all updates are done within a transaction
+        """
+        catalogue_items = self._catalogue_items_collection.find({}, session=session)
+
+        logger.info("expected_lifetime forward migration")
+        for catalogue_item in catalogue_items:
+            try:
+                old_item = OldCatalogueItemOut(**catalogue_item)
+
+                new_item_data = {
+                    "catalogue_category_id": old_item.catalogue_category_id,
+                    "manufacturer_id": old_item.manufacturer_id,
+                    "name": old_item.name,
+                    "description": old_item.description,
+                    "cost_gbp": old_item.cost_gbp,
+                    "cost_to_rework_gbp": old_item.cost_to_rework_gbp,
+                    "days_to_replace": old_item.days_to_replace,
+                    "days_to_rework": old_item.days_to_rework,
+                    "drawing_number": old_item.drawing_number,
+                    "drawing_link": old_item.drawing_link,
+                    "expected_lifetime": None,
+                    "item_model_number": old_item.item_model_number,
+                    "is_obsolete": old_item.is_obsolete,
+                    "obsolete_reason": old_item.obsolete_reason,
+                    "obsolete_replacement_catalogue_item_id": old_item.obsolete_replacement_catalogue_item_id,
+                    "notes": old_item.notes,
+                    "properties": old_item.properties,
+                    "created_time": catalogue_item["created_time"],
+                    "modified_time": catalogue_item["modified_time"],
+                }
+
+                new_item = NewCatalogueItemIn(**new_item_data)
```

Comment:
You can use `.model_dump()` on the old model as well e.g.
```
new_item = NewCatalogueItemIn(**old_item.model_dump(), expected_lifetime=None)
```

Then you can use something like
```
update_data = {**new_item.model_dump(), "modified_time":  old_item.modified_time}